src/khoj/processor/conversation/gpt.py

```diff
@@ -102,86 +82,38 @@ def extract_questions(text, model="text-davinci-003", conversation_log={}, api_k
     current_new_year = today.replace(month=1, day=1)
     last_new_year = current_new_year.replace(year=today.year - 1)
 
-    prompt = f"""
-You are Khoj, an extremely smart and helpful search assistant with the ability to retrieve information from the users notes.
-- The user will provide their questions and answers to you for context.
-- Add as much context from the previous questions and answers as required into your search queries.
-- Break messages into multiple search queries when required to retrieve the relevant information.
-- Add date filters to your search queries from questions and answers when required to retrieve the relevant information.
-
-What searches, if any, will you need to perform to answer the users question?
-Provide search queries as a JSON list of strings
-Current Date: {today.strftime("%A, %Y-%m-%d")}
-
-Q: How was my trip to Cambodia?
-
-["How was my trip to Cambodia?"]
-
-A: The trip was amazing. I went to the Angkor Wat temple and it was beautiful.
-
-Q: Who did i visit that temple with?
-
-["Who did I visit the Angkor Wat Temple in Cambodia with?"]
-
-A: You visited the Angkor Wat Temple in Cambodia with Pablo, Namita and Xi.
-
-Q: What national parks did I go to last year?
-
-["National park I visited in {last_new_year.strftime("%Y")} dt>=\\"{last_new_year.strftime("%Y-%m-%d")}\\" dt<\\"{current_new_year.strftime("%Y-%m-%d")}\\""]
-
-A: You visited the Grand Canyon and Yellowstone National Park in {last_new_year.strftime("%Y")}.
-
-Q: How are you feeling today?
-
-[]
-
-A: I'm feeling a little bored. Helping you will hopefully make me feel better!
-
-Q: How many tennis balls fit in the back of a 2002 Honda Civic?
-
-["What is the size of a tennis ball?", "What is the trunk size of a 2002 Honda Civic?"]
-
-A: 1085 tennis balls will fit in the trunk of a Honda Civic
-
-Q: Is Bob older than Tom?
-
-["When was Bob born?", "What is Tom's age?"]
-
-A: Yes, Bob is older than Tom. As Bob was born on 1984-01-01 and Tom is 30 years old.
-
-Q: What is their age difference?
-
-["What is Bob's age?", "What is Tom's age?"]
-
-A: Bob is {current_new_year.year - 1984 - 30} years older than Tom. As Bob is {current_new_year.year - 1984} years old and Tom is 30 years old.
-
-{chat_history}
-Q: {text}
-
-"""
+    prompt = prompts.extract_questions.format(
+        current_date=today.strftime("%A, %Y-%m-%d"),
+        last_new_year=last_new_year.strftime("%Y"),
+        last_new_year_date=last_new_year.strftime("%Y-%m-%d"),
+        current_new_year_date=current_new_year.strftime("%Y-%m-%d"),
+        bob_tom_age_difference={current_new_year.year - 1984 - 30},
+        bob_age={current_new_year.year - 1984},
```

Comment:
What are these two values for?

Reply:
These values are used to dynamically populate the prompt with examples for GPT to get a better sense of how we expect it to extract questions. These specific variables are used for a q&q example where we ask GPT the age difference between Bob and Tom and it needs to ask multiple (2) questions "What is Tom's age", "What is Bob's age" (and later form a response based on the search results). See `prompts.py:extract_questions` for usage